### ISSUE TITLE
Intersect of the same polygons returns rotated vertices

### ISSUE THREAD

**Issue by klaucos** [shapely/shapely#1254]
## Expected behavior and actual behavior.

In a previous version of shapely(1.6.4.post2) we were using intersecting a pair of polygons. 
In case polygon was intersected over same polygon/itself it returned exactly the same geometry.
With update to 1.8.0 we have encountered weird behavior where every intersect rotates vertices by one and polygon is not exactly the same anymore because of this change. 
I'd expect that intersect of itself should provide the same geometry or at least return predictable output.

## Steps to reproduce the problem.

```
from shapely.geometry import Polygon

extent = Polygon([
    (10.55498600006104, 42.94454893489702),
    (10.56210994720459, 42.93694613058707),
    (10.54811954498291, 42.93672620078932),
    (10.55498600006104, 42.94454893489702)
])

intersect = extent.intersection(extent)
if extent != intersect:
    print('Not the same')
intersect2 = intersect.intersection(intersect)
if intersect2 != intersect:
    print('Still not the same')
intersect3 = intersect2.intersection(intersect2)
if intersect3 != intersect2:
    print('Still not the same')
print(extent.wkt)
# POLYGON ((10.55498600006104 42.94454893489702, 10.56210994720459 42.93694613058707, 10.54811954498291 42.93672620078932, 10.55498600006104 42.94454893489702))
print(intersect.wkt)
# POLYGON ((10.56210994720459 42.93694613058707, 10.54811954498291 42.93672620078932, 10.55498600006104 42.94454893489702, 10.56210994720459 42.93694613058707))
print(intersect2.wkt)
# POLYGON ((10.54811954498291 42.93672620078932, 10.55498600006104 42.94454893489702, 10.56210994720459 42.93694613058707, 10.54811954498291 42.93672620078932))
print(intersect3.wkt)
# POLYGON ((10.55498600006104 42.94454893489702, 10.56210994720459 42.93694613058707, 10.54811954498291 42.93672620078932, 10.55498600006104 42.94454893489702))
# We got bact to original extent.
assert intersect3 == extent
```

## Operating system

Ubuntu 20.1

## Shapely version and provenance

1.8.0 from pip, Python 3.8.10

**Comment by mwtoews** [comment]
This is perhaps a strange quirk of [OverlayNG](http://lin-ear-th-inking.blogspot.com/2020/06/jts-overlayng-noding-strategies.html) in GEOS 3.9 (@dr-jts ?), which was bundled with binary wheels for shapely 1.8.

As a more reliable workaround, a new function with shapely 1.8 is `.normalize()`:
```python
assert intersect.normalize() == intersect2.normalize() == intersect3.normalize()
```

**Comment by dr-jts** [comment]
> This is perhaps a strange quirk of [OverlayNG](http://lin-ear-th-inking.blogspot.com/2020/06/jts-overlayng-noding-strategies.html) in GEOS 3.9 (@dr-jts ?)

Yes, this is a quirk of the OverlayNG codebase.  Strictly speaking OverlayNG does not provide any guarantees about output coordinate order.  But this is sufficiently odd that perhaps it worth looking into to see if this can be remediated somehow.  And yes, using `normalize` is always a good idea if the coordinate order in rings needs to be relied upon.

**Comment by connorferster** [comment]
We have also encountered this when working on [sectionproperties.](https://github.com/robbievanleeuwen/section-properties/tree/v2). It made unit test writing a bit troublesome.

**Comment by sgillies** [comment]
I marked this as an upstream issue and will close.

**Comment by dbaston** [comment]
Fixed in GEOS 3.15, https://github.com/libgeos/geos/pull/1412
